airgap/v2/pkg/dqlite/driver/dqlite.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021 IBM Corp.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dqlite
+
+import (
+	"database/sql"
+	"strconv"
+	"strings"
+
+	_ "github.com/mattn/go-sqlite3"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/callbacks"
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/logger"
+	"gorm.io/gorm/migrator"
+	"gorm.io/gorm/schema"
+)
+
+// DriverName is the default driver name for DQLite.
+const DriverName = "dqlite"
+
+type Dialector struct {
+	db        *sql.DB
+	dialector *sqlite.Dialector
+}
+
+func Open(db *sql.DB) gorm.Dialector {
+	return &Dialector{db: db, dialector: &sqlite.Dialector{}}
+}
+
+func (wrapper Dialector) Name() string {
+	return DriverName
+}
+
+func (wrapper Dialector) Initialize(db *gorm.DB) (err error) {
+	if wrapper.dialector.DriverName == "" {
+		wrapper.dialector.DriverName = DriverName
+	}
+
+	// register callbacks
+	callbacks.RegisterDefaultCallbacks(db, &callbacks.Config{
+		LastInsertIDReversed: true,
+	})
+
+	if wrapper.dialector.Conn != nil {
+		db.ConnPool = wrapper.dialector.Conn
+	} else {
+		db.ConnPool = wrapper.db
+	}
+
+	for k, v := range wrapper.dialector.ClauseBuilders() {
+		db.ClauseBuilders[k] = v
+	}
+	return
+}
+
+func (wrapper Dialector) ClauseBuilders() map[string]clause.ClauseBuilder {
+	return map[string]clause.ClauseBuilder{
```

Comment:
You can call the dialector on the struct.
```suggestion
	return wrapper.dialector.ClauseBuidlers()
```

Reply:
Please update the other functions in this driver to use the underlying dialector on the struct. That's why it's a wrapper.